### DISCUSSION TITLE
How do you all manage git credentials?

### DISCUSSION THREAD

**Discussion by rernest7** [pvtl/docker-dev#32] [Q&A] [answered]
Hi everyone,
I wonder How do you all manage credentials for github and so on?
For the past 2 days I'm trying to setup github credentials so I won't have to enter my token every time I'm cloning private repos.
I'm using personal tokens and the last, naive attempt was: I  added token to .env, and then did the following in custom_scripts, but it doesnt create .git-credentials file and well it just doesn't seem right, but I'm out of options:
git config --global credential.helper store
echo "https://${GITHUB_TOKEN}@github.com" | sudo tee -a /home/${CUSTOM_USER_NAME}/.git-credentials
Sorry for my ignorance, but I don't know what's going wrong, can someone please guide me in the right direction?
Thanks

**Comment by wireblue** [comment, marked as answer]
We normally interact with Git on our host system, outside of the Docker containers. We find it simpler and faster.

If you have a specific use-case that requires Git inside the containers then I'm sure there is a way to pass an environment variable through or something. Sorry, it's not really something we'd needed to do in the past!

**Reply by rernest7** [reply]
I agree - It is faster and easier! 
The git inside the container made me assume that it's the better/right way.
That said, thanks for sharing and bringing back my sanity :) oh and thanks for the amazing project 💘

**Reply by rernest7** [reply]
@wireblue if I may, I have a quick side question: How do you handle your CI pipeline? Do you host it on a separate container, or host?

**Reply by wireblue** [reply]
We typically use Docker for local development, but separate servers for staging and production.
eg. For Laravel sites, we typically prefer Laravel Forge.

**Reply by rernest7** [reply]
Hi, I was stupid in my last question, i apologize for that. What i meant to ask is how do you do git hooks (husky, etc.)?

**Reply by wireblue** [reply]
No, we don't use any Git hooks at the moment.
We occasionally use Github Actions though.

**Reply by rernest7** [reply]
Thanks <3
